src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -1342,7 +1342,19 @@ def ClearTempTables(message: proto.Request) -> None:
 
 def base64_str_to_request(base64_str: str) -> proto.Request:
```

Comment:
I think we should be here a bit more careful with API naming and semantics.

Technically, `"\n"` is not a valid character in base64. The function itself right now does exactly one thing, which is taking a valid base64_str (for which the user is responsible, we can do a check) and decodes it. I would suggest to keep the basic function as is, but for convenience add a new one which does the repeated decode and then merges the result together.

I.e.,
```
def base64_str_to_request(base64_str: str) -> proto.Request:
  ....
 < add check that characters are only in `[0-9+/=a-zA-Z]` which is base64 encoding>
 
 def base64_lines_to_requst(base64_lines: str) -> proto.Request:
    # allow here character "\n"
    requests = [base64_str_to_request]
    return merge_requests(requests)
```

This is I think anyways the logic you have in mind here, but easier to understand

Reply:
Updated